core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -2692,7 +2693,7 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=t
 			$protected = $blnProtected;
 
 			// Check whether the folder is public
-			if ($protected === true && \in_array('.public', $content))
+			if ($protected === true && is_file(Path::join($folders[$f], '.public')))
```

Comment:
Note that this will increase the amount of file system access operations though.

Reply:
Changing it to use `!is_dir()` could help in this case because that call should already be cached from the code above.

Reply:
Good point! Changed in 3cc318964

Reply:
Alternatively, we could disallow naming a folder `.public` from the UI, make it a known limitation and just keep the exceptions in case someone perfomed the operation on the filesystem nevertheless.

Reply:
But the system still needs to know whether it is a file or not, otherwise the folder will still be displayed as being public, when it is not.

Reply:
Yes, this would only work if a `.public` folder was implicitly considered public.